src/app/modules/item/pages/item-log-view/item-log-view.component.ts

```diff
@@ -51,16 +40,51 @@ export class ItemLogViewComponent implements OnChanges {
   ) {}
 
   ngOnChanges(): void {
-    if (this.itemData) this.id$.next(this.itemData.item.id);
+    if (!this.itemData) {
+      return;
+    }
+
+    this.id$.next(this.itemData.item.id);
+    this.logColumns = this.getLogColumns(this.itemData.item.type);
   }
 
-  getData$(id: string): Observable<Data> {
+  private getData$(id: string): Observable<Data> {
     return this.activityLogService.getActivityLog(id).pipe(
       map((data: ActivityLog[]) => ({
-        columns: logColumns,
+        columns: this.logColumns || [],
         rowData: data
       }))
     );
   }
 
+  private getLogColumns(type: ItemType): Column[] {
```

Comment:
With a function called `getLogColumns` you just expect to get the current content of the `logColumns` variable. (`get<varname>` is the default accesor in many languages). I would prefer a name like `columnForType(type: ItemType)`.

Reply:
Fixed

Reply:
Fixed